app/components/ui/FormEdit.tsx

```diff
@@ -0,0 +1,55 @@
+"use client";
+import React from "react";
+import { formType } from "@/app/schemas/formSchema";
+import {
+  Accordion,
+  AccordionContent,
+  AccordionItem,
+  AccordionTrigger,
+} from "@/components/ui/accordion";
+import { HouseType } from "../../models/Property";
+
+interface Props {
+  formData: formType;
+}
+
+function mapHouseType(houseType: HouseType) {
+  if (houseType === "D") {
+    return "Detached";
+  }
+
+  if (houseType === "S") {
+    return "Semi-detached";
+  }
+
+  if (houseType === "T") {
+    return "Terrace";
+  }
+
+  if (houseType === "F") {
+    return "Flat";
+  }
+}
```

Comment:
```suggestion
const houseTypeDisplayNames: Record<HouseType, string> = {
  D: "Detached",
  S: "Semi-detached",
  T: "Terrace",
  F: "Flat",
};
```

And then later - `houseTypeDisplayNames[formData.houseType]`

Using a plain object for a map instead of a function is probably simpler, more maintainable, and more type safe (adding a new house type will throw an error here, as will missing one out).